ci/delete_thin-edge/delete_thin_edge.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF
+USAGE:
+    delete_thin_edge [remove/purge]
+EOF
+}
+
+stop_a_service_if_running() {
+    local status
+    status=$(systemctl show -p ActiveState --value "$1")
+    if [ "$status" == "active" ]; then
+        sudo systemctl stop "$1"
+    fi
+}
+
+stop_services() {
+    stop_a_service_if_running "tedge-watchdog.service"
+    stop_a_service_if_running "tedge-mapper-collectd.service"
+    stop_a_service_if_running "c8y_log_plugin"
+    stop_a_service_if_running "c8y_configuration_plugin"
+    stop_a_service_if_running "apama"
+}
+
+remove_or_purge_package_if_exists() {
+    local status
+    status=$(dpkg -s "$2" | grep -w installed) && returncode=$? || returncode=$?
+    if [ "$status" == "Status: install ok installed" ]; then
+        sudo apt --assume-yes "$1" "$2"
+    fi
+}
+
+remove_packages() {
+    remove_or_purge_package_if_exists "remove" "tedge"
+    remove_or_purge_package_if_exists "remove" "tedge_apt_plugin"
+    remove_or_purge_package_if_exists "remove" "tedge_apama_plugin"
+    remove_or_purge_package_if_exists "remove" "c8y_log_plugin"
+    remove_or_purge_package_if_exists "remove" "c8y_configuration_plugin"
+}
+
+disconnect_if_connected_to_cloud() {
+    if [ -f "/etc/tedge/mosquitto-conf/$1-bridge.conf" ]; then
+        sudo tedge disconnect "$1"
+    fi
+}
+
+disconnect_from_cloud() {
+    disconnect_if_connected_to_cloud "c8y"
+    disconnect_if_connected_to_cloud "az"
+}
+
+remove_thin_edge_io() {
+    echo "remove thin_edge_io"
+    disconnect_from_cloud
+    stop_services
+    remove_packages
+}
+
+purge_thin_edge_io() {
+    echo "purge thin-edge-io"
+    remove_or_purge_package_if_exists "purge" "tedge"
+    remove_or_purge_package_if_exists "purge" "tedge_apt_plugin"
+    remove_or_purge_package_if_exists "purge" "tedge_apama_plugin"
+    remove_or_purge_package_if_exists "purge" "c8y_log_plugin"
+    remove_or_purge_package_if_exists "purge" "c8y_configuration_plugin"
+    sudo DEBIAN_FRONTEND=noninteractive apt --assume-yes purge mosquitto-clients mosquitto libmosquitto1 collectd-core collectd
```

Comment:
Purging other software is too much. Remove this line.

Reply:
It is mentioned in the ticket, so I added it here.

Reply:
Murat just copied the commands which we use to "clean up" Raspberry Pi to fit our use-case. The script is helping user to remove `thin-edge.io`, however, better doing less than overdoing. I don't think it is good that uninstalling collectd even if the user has already installed it before the installation of thin-edge.